src/ccstruct/imagedata.cpp

```diff
@@ -546,6 +547,31 @@ bool DocumentData::ReCachePages() {
     delete page;
   }
   pages_.clear();
+#if !defined(TESSERACT_IMAGEDATA_AS_PIX)
+  auto name_size = document_name_.size();
+  if (name_size > 4 && document_name_.substr(name_size - 4) == ".png") {
```

Comment:
Why not use [to use std::filesystem::path](https://en.cppreference.com/w/cpp/filesystem/path/extension) as in https://github.com/tesseract-ocr/tesseract/blob/b217cfcfa3b5c41e20414428e5d4700af130b192/src/training/unicharset_extractor.cpp#L74

Reply:
Path refactoring should be done as more general procedure. It is ok to keep std::string in very local code.

Reply:
So would you prefer using `std::filesystem::path::extension()` here or not?

Reply:
It's ok as is. fs::paths can be refactored for 6.0 wit api changes/breakage.

Reply:
@egorpugin: `document_name_` is not part of tesseract API - it is the private object used in imagedata.cpp only, So I do not think it will break something or change API.

@stweil: I am fine with the current PR code, just I think we should step by step use c++17 features.